File: DSA/LakshaKundra_590013237/Day_37_Question_3.py
def delete_middle(stack, k):
    if k == 1:
        stack.pop()
        return
    temp = stack.pop()
    delete_middle(stack, k - 1)
    stack.append(temp)

def remove_middle(stack):
    k = (len(stack) + 1) // 2
    delete_middle(stack, k)
    return stack

print(remove_middle([1, 2, 3, 4, 5]))   # Output: [1, 2, 4, 5]
print(remove_middle([10, 20, 30]))      # Output: [10, 30]

File: DSA/Gautam_Shyamnani_590013078/Day37Question3.cpp
#include <iostream>
#include <stack>
using namespace std;


void deleteMiddle(stack<int>& s, int currentIndex, int middleIndex) {
    
    if (currentIndex == middleIndex) {
        s.pop(); 
        return;
    }

    
    int top = s.top();
    s.pop();

    deleteMiddle(s, currentIndex + 1, middleIndex);

   
    s.push(top);
}

void removeMiddle(stack<int>& s) {
    int size = s.size();
    if (size == 0) return;

    int middleIndex = size / 2; 
    deleteMiddle(s, 0, middleIndex);
}

int main() {
    stack<int> s;
    s.push(1);
    s.push(2);
    s.push(3);
    s.push(4);
    s.push(5);

    cout << "Original stack: ";
    stack<int> temp = s;
    while (!temp.empty()) {
        cout << temp.top() << " ";
        temp.pop();
    }
    cout << endl;

    removeMiddle(s);

    cout << "After deleting middle element: ";
    while (!s.empty()) {
        cout << s.top() << " ";
        s.pop();
    }
    cout << endl;

    return 0;
}
